aws-rds-dbclusterendpoint/src/main/java/software/amazon/rds/dbclusterendpoint/BaseHandlerStd.java

```diff
@@ -0,0 +1,139 @@
+package software.amazon.rds.dbclusterendpoint;
+
+import software.amazon.awssdk.services.rds.RdsClient;
+import software.amazon.awssdk.services.rds.model.DBClusterEndpoint;
+import software.amazon.awssdk.services.rds.model.DbClusterEndpointAlreadyExistsException;
+import software.amazon.awssdk.services.rds.model.DbClusterEndpointNotFoundException;
+import software.amazon.awssdk.services.rds.model.DbClusterEndpointQuotaExceededException;
+import software.amazon.awssdk.services.rds.model.DbClusterNotFoundException;
+import software.amazon.awssdk.services.rds.model.DescribeDbClusterEndpointsResponse;
+import software.amazon.cloudformation.exceptions.CfnNotFoundException;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.HandlerErrorCode;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ProxyClient;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+import software.amazon.cloudformation.proxy.delay.Constant;
+import software.amazon.rds.common.error.ErrorRuleSet;
+import software.amazon.rds.common.error.ErrorStatus;
+import software.amazon.rds.common.handler.Commons;
+import software.amazon.rds.common.handler.HandlerConfig;
+import software.amazon.rds.common.handler.Tagging;
+
+import java.time.Duration;
+import java.util.Optional;
+
+public abstract class BaseHandlerStd extends BaseHandler<CallbackContext> {
+
+    protected static final String DB_CLUSTER_ENDPOINT_AVAILABLE = "available";
+    protected static final String CUSTOM_ENDPOINT = "CUSTOM";
+
+    protected static final Constant BACKOFF_DELAY = Constant.of()
+            .timeout(Duration.ofSeconds(180L))
+            .delay(Duration.ofSeconds(15L))
+            .build();
+    protected static final ErrorRuleSet DEFAULT_DB_CLUSTER_ENDPOINT_ERROR_RULE_SET = ErrorRuleSet.builder()
+            .withErrorClasses(ErrorStatus.failWith(HandlerErrorCode.AlreadyExists),
+                    DbClusterEndpointAlreadyExistsException.class)
+            .withErrorClasses(ErrorStatus.failWith(HandlerErrorCode.NotFound),
+                    DbClusterEndpointNotFoundException.class,
+                    DbClusterNotFoundException.class)
+            .withErrorClasses(ErrorStatus.failWith(HandlerErrorCode.ServiceLimitExceeded),
+                    DbClusterEndpointQuotaExceededException.class)
```

Comment:
Can we add all exception that mentioned in documentation  i.e: `DBInstanceNotFound`?
Otherwise those exception will be mapped to internal failure. They are possible errors can happen during creation/modification of `ClusterEndpoint` resource.
ref: https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_CreateDBClusterEndpoint.html